Khan/wonder-blocks
# WB-1638: Fix tabbing in popover

## Summary:
There is a bug in the popover where tabbing out of the popover (while it
is still open) will cause the focus order to be incorrect. More
specifically, the focus will be returned to the trigger element, while
it should be returned to the next element in the tab order.

This is because the popover is not properly handling the case where the
trigger element is the last element in the tab order. This PR fixes that
by making all the focusable elements in the popover non-tabbable, and
then make them tabbable again when the popover gains focus.

For more reference on the actual error, see: https://khanacademy.atlassian.net/browse/CL-1259

https://github.com/Khan/wonder-blocks/assets/843075/ca72e266-9091-409e-9065-3754e4c681c9

Issue: WB-1638

## Test plan:

- Navigate to /?path=/story/popover-popover--keyboard-navigation
- Start tabbing through the elements in the story
- Verify that the focus order is correct
- Open the popover
- Tab through the elements in the popover and verify that the focus
  order is still correct
- Tab backwards through the elements in the popover and verify that the
  focus order is still correct (using shift+tab).
- Close the popover and verify that the focus order is still correct


https://github.com/Khan/wonder-blocks/assets/843075/a8831575-84a8-4bb3-a0f3-11a26d872076
